### PR TITLE
Add micrometer tests

### PR DESCRIPTION
**Pull request by fabiobrz**
Backporting tests in https://github.com/wildfly/wildfly/pull/17898 in order to be able and test Micrometer integration in JBoss EAP XP 5.

- Internal validation job reference: **eap-8.x-microprofile-simple-face**, run nr. **110** :runner: 

---

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] ~Link~ Reference to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public static final int DOCKER_CONTAINER_PORT_JAEGER_QUERY = 16686;
+    public static final int DOCKER_HOST_PORT_JAEGER_QUERY = DOCKER_CONTAINER_PORT_JAEGER_QUERY;
+    public static final int DOCKER_CONTAINER_PORT_JAEGER_OTLP = OpenTelemetryCollectorContainer.DEFAULT_OTLP_GRPC_PORT;
+    public static final int DOCKER_HOST_PORT_JAEGER_OTLP = DOCKER_CONTAINER_PORT_JAEGER_OTLP - 100;
+    private String jaegerEndpoint;
+
+    private final Docker jaeger;
+
+    private JaegerContainer() {
+        jaeger = new Docker.Builder("jaeger", "quay.io/jaegertracing/all-in-one:latest")
```
Comment (marekkopecky) [inline micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java:27]: Does it make sense to freeze the version? Maybe 1.58?

**Comment by marekkopecky** [comment]
I'm not much happy about the javadoc (missing/poor) of classes copy-pasted from wildfly. But my understanding is that this is temporary solution, so I'm ok with it now.

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.jboss.eap.qe.micrometer.container.PrometheusMetric;
+import org.jboss.eap.qe.ts.common.docker.Docker;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+
+public abstract class BaseMultipleTestCase {
```
Comment (marekkopecky) [inline micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java:17]: My understanding is that this class is not copy pasted from WF, right? So maybe can you please add some short javadoc here?

**Comment by marekkopecky** [comment]
I added just a few minor comments, otherwise LGTM, thank you!

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java (hunk 2 above)
Comment (fabiobrz) [reply micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java:17]: All of them are all copied from WildFly, but I might add some headers and Javadoc as well. Fixing.

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java (hunk 1 above)
Comment (fabiobrz) [reply micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java:27]: Yeah, we're aiming at having latest for runtimes we test interoperability with.
But this is a tooling component so I do agree at freezing the version. Fixing, thanks.

**Comment by fabiobrz** [comment]
> I added just a few minor comments, otherwise LGTM, thank you!

Should be fixed now, thanks @marekkopecky !

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java (hunk 1 above)
Comment (marekkopecky) [reply micrometer/src/test/java/org/jboss/eap/qe/micrometer/container/JaegerContainer.java]: Thank you!

File: micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java (hunk 2 above)
Comment (marekkopecky) [reply micrometer/src/test/java/org/jboss/eap/qe/micrometer/multiple/BaseMultipleTestCase.java:21]: Thank you!

**Pull request merged**